src/lib/containers/Goals.tsx

```diff
@@ -0,0 +1,163 @@
+import React, { useEffect, useState } from 'react'
+import {
+  QueryResultBundle,
+  QueryBundleRequest,
+  FileHandleAssociation,
+  FileHandleAssociateType,
+  BatchFileRequest,
+} from '../utils/synapseTypes'
+import { SynapseClient, SynapseConstants } from '../utils'
+import { SynapseClientError, getFiles } from '../utils/SynapseClient'
+import { Error } from '../containers/Error'
+import QueryCount from './QueryCount'
+
+export type GoalsProps = {
+  entityId: string
+  token?: string
+}
+
+export enum ExpectedColumns {
+  TABLEID = 'TableId',
+  TITLE = 'Title',
+  SUMMARY = 'Summary',
+  LINK = 'Link',
+  ASSET = 'Asset',
+}
+
+export default function (props: GoalsProps) {
+  const { entityId, token } = props
+
+  const [queryResult, setQueryResult] = useState<
+    QueryResultBundle | undefined
+  >()
+  const [error, setError] = useState<SynapseClientError | undefined>()
+  const [assets, setAssets] = useState<string[] | undefined>()
+
+  const getFieldIndex = (
+    name: string,
+    result: QueryResultBundle | undefined,
+  ) => {
+    return (
+      result?.columnModels?.findIndex(el => {
+        return el.name === name
+      }) ?? -1
+    )
+  }
+
+  const isValidData = (queryResult: QueryResultBundle) => {
+    return (
+      getFieldIndex(ExpectedColumns.TABLEID, queryResult) !== -1 &&
+      getFieldIndex(ExpectedColumns.TITLE, queryResult) !== -1 &&
+      getFieldIndex(ExpectedColumns.SUMMARY, queryResult) !== -1 &&
+      getFieldIndex(ExpectedColumns.LINK, queryResult) !== -1 &&
+      getFieldIndex(ExpectedColumns.ASSET, queryResult) !== -1
+    )
+  }
+
+  useEffect(() => {
+    const getData = async () => {
+      const request: QueryBundleRequest = {
+        concreteType: 'org.sagebionetworks.repo.model.table.QueryBundleRequest',
+        entityId,
+        partMask:
+          SynapseConstants.BUNDLE_MASK_QUERY_COLUMN_MODELS |
+          SynapseConstants.BUNDLE_MASK_QUERY_RESULTS,
+        query: {
+          sql: `SELECT * FROM ${entityId}`,
```

Comment:
I think these need an explicit order (controlled by the portal stakeholder).

Reply:
You can control the order of the rows right? I created a table and the rows appear to be in insert order?

Reply:
the alternative is we take in sql with an orderBy clause